### PR TITLE
Preserve the type of application builder in chained extension methods

### PR DESCRIPTION
**Pull request by sliekens**
This is a slight improvement to the usability of extensions in the ServiceDefaults project template. Previously, you couldn't chain `WebApplication.CreateBuilder` and the `ServiceDefaults` extensions.

For example, this code won't compile:

``` csharp
var builder = WebApplication.CreateBuilder(args)
    .AddServiceDefaults();

var app = builder.Build();
               // ^^^^^
               // CS1061: 'IHostApplicationBuilder' does not contain a definition for 'Build' and no accessible extension method 'Build' accepting a first argument of type 'IHostApplicationBuilder' could be found
```

After this change, compilation will succeed.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4759)

### CODE REVIEW

**Comment by eerhardt** [comment]
@mitchdenny @DamianEdwards @davidfowl - thoughts here?

If we take this change, we need to update ServiceDefaults everywhere - playground apps, all the templates, etc.

**Comment by mitchdenny** [comment]
I don't have a problem with this change, but the PR will need to include the fixes to playground apps etc as @eerhardt has identified.

I would defer to folks who have had to deal with the reality of project template changes over time and what issues that can introduce though.

**Comment by DamianEdwards** [comment]
This seems like something we should consider for 9.0

**Comment by mitchdenny** [comment]
@sliekens did you want to add the playground app updates here too?

**Comment by sliekens** [comment]
Yeah I realized after creating the PR that the ServiceDefaults template was not the only place to make changes, though I'm not sure how many other projects would need to be changed.

**Comment by mitchdenny** [comment]
If you look under /playground you'll see that we generally share a single service defaults project there.

**Comment by sliekens** [comment]
@mitchdenny I'm noticing other kinds of drift between the different copies of the Extensions.cs, is it okay if I re-align them while I'm making this change?

File: src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs [hunk 1]
```diff
@@ -31,10 +32,16 @@ public static IHostApplicationBuilder AddServiceDefaults(this IHostApplicationBu
             http.AddServiceDiscovery();
         });
 
+        // Uncomment the following to restrict the allowed schemes for service discovery.
+        // builder.Services.Configure<ServiceDiscoveryOptions>(options =>
+        // {
+        //     options.AllowedSchemes = ["https"];
+        // });
+
```
Comment (DamianEdwards) [inline src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs:40]: Hmm where did this come from? I'm not seeing it in the current templates. Did we have this and then remove it at some point?

**Review by DamianEdwards — CHANGES_REQUESTED** [review]
I'm supportive of this change in the templates, but the remaining playground apps need to be updated and the builds errors regarding the unnecessary `using` directive need to be fixed.

File: src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs (hunk 1 above)
Comment (DamianEdwards) [reply src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs:40]: Ah got it, thanks. It's fine to add it to the other templates then, but we don't need it in the playground apps.

Comment (sliekens) [reply src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs:40]: OK, it's now removed from the playground apps

File: tests/TestingAppHost1/TestingAppHost1.ServiceDefaults/Extensions.cs [hunk 2]
```diff
@@ -1,6 +1,3 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (DamianEdwards) [inline tests/TestingAppHost1/TestingAppHost1.ServiceDefaults/Extensions.cs:1]: You need to add this header back, it's causing a build failure.

**Review by DamianEdwards — CHANGES_REQUESTED** [review]
Seems you inadvertently deleted the required header from one of the files.

File: tests/TestingAppHost1/TestingAppHost1.ServiceDefaults/Extensions.cs (hunk 2 above)
Comment (sliekens) [reply tests/TestingAppHost1/TestingAppHost1.ServiceDefaults/Extensions.cs]: Oh man, I put it back and I'm gonna watch for the build to finish successfully this time 🤣.

**Review by DamianEdwards — APPROVED** [review]
Thank you!

**Pull request merged**